packages/legend-studio/src/components/editor/edit-panel/connection-editor/DatabaseBuilderEditor.tsx

```diff
@@ -0,0 +1,347 @@
+/**
+ * Copyright (c) 2020-present, Goldman Sachs
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import { observer } from 'mobx-react-lite';
+import Dialog from '@material-ui/core/Dialog';
+import type { TreeNodeContainerProps } from '@finos/legend-studio-components';
+import {
+  PanelLoadingIndicator,
+  SaveIcon,
+  FireIcon,
+  clsx,
+  TreeView,
+} from '@finos/legend-studio-components';
+import { ReflexContainer, ReflexElement, ReflexSplitter } from 'react-reflex';
+import { useEffect } from 'react';
+import type {
+  DatabaseBuilderState,
+  DatabaseBuilderTreeData,
+  DatabaseBuilderTreeNodeData,
+} from '../../../../stores/editor-state/element-editor-state/connection/DatabaseBuilderState';
+import {
+  ColumnDatabaseBuilderTreeNodeData,
+  SchemaDatabaseBuilderTreeNodeData,
+  TableDatabaseBuilderTreeNodeData,
+} from '../../../../stores/editor-state/element-editor-state/connection/DatabaseBuilderState';
+import {
+  FaCircle,
+  FaCheckCircle,
+  FaChevronDown,
+  FaChevronRight,
+  FaRegCircle,
+} from 'react-icons/fa';
+import { capitalize } from '@finos/legend-studio-shared';
+import { SchemaIcon, TableIcon } from '../../../shared/Icon';
+import { TextInputEditor } from '../../../shared/TextInputEditor';
+import { EDITOR_LANGUAGE } from '../../../../stores/EditorConfig';
+import {
+  generateColumnTypeLabel,
+  renderColumnTypeIcon,
+} from '../../../../stores/editor-state/element-editor-state/mapping/relational/DatabaseHelper';
+
+const getNodeIcon = (node: DatabaseBuilderTreeNodeData): React.ReactNode => {
+  if (node instanceof SchemaDatabaseBuilderTreeNodeData) {
+    return <SchemaIcon />;
+  } else if (node instanceof TableDatabaseBuilderTreeNodeData) {
+    return <TableIcon />;
+  } else if (node instanceof ColumnDatabaseBuilderTreeNodeData) {
+    return renderColumnTypeIcon(node.column.type);
+  }
+  return null;
+};
+
+const QueryBuilderGraphFetchTreeNodeContainer: React.FC<
+  TreeNodeContainerProps<
+    DatabaseBuilderTreeNodeData,
+    {
+      toggleCheckedNode: (node: DatabaseBuilderTreeNodeData) => void;
+      isPartiallySelected: (node: DatabaseBuilderTreeNodeData) => boolean;
```

Comment:
is there a reason we don't make this an attribute of the node instead of a function?

Reply:
its calculated based on the current db and the isChecked flag so doesn't really belong directly on the node.

Reply:
makes sense!